game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -341,6 +342,18 @@ public GameData getData() {
     return gameData;
   }
 
+  public boolean getEditMode() {
+    final boolean isEditMode;
+    // use GameData from mapPanel since it will follow current history node
```

Comment:
I guess this comment does no longer make 100% sense (if it ever did).

One thing that I think is weird with this change in general:
Even if the method itself seems to delegate 50% of it's method to the MapPanel class, the only usages are part of TripleAFrame which seems counter-intuitive to me 🤔

Reply:
I'm confused. This method delegates basically everything to MapPanel. The `getEditMode` method being in `TripleAFrame` seems to be classic feature envy code smell. The fact `TripleAFrame` is the only consumer, does not mean it should also know/manipulate the internal data state of `MapPanel` by getting internal data and then managing and interpretting that data state.

THe comment still kinda makes sense perhaps. MapPanel, when in history mode, gets a clone of the game data. It's subtle and hard to really grok the data state of MapPanel (not well encapsulated at all, but also MapPanel is so bloated, it's hard to keep track of the data state anyways)

Reply:
Good point about the encapsulation.
If the 50% was what confused you: Just an arbitrary number, didn't want to write 100% because that's not really accurant either